File: Model/World/GrassTile.kt
package Model.World

import Model.Abstract.Tile
import Model.Location
import Model.Map
import Model.Move

class GrassTile(loc: Location, occupied: Boolean): Tile() {
    override var loc = loc
    override var occ = occupied
    override var use = false
    override var drive = false
    override var traverse = true
    override var owner = 0

    override fun isValidMove(move: Move, map: Map): Boolean {
        return false
    }
}

File: typescript/src/PlanetTS1/Model/World/GrassTile.ts
import { ALocation as ALocation } from "../Utilities"

export class GrassTile extends RealTile {
    loc : ALocation
    occ : boolean
    use : boolean
    drive : boolean
    traverse : boolean
    owner : number
    icon : string
    minCost : number
    gasCost : number
    nrgCost : number
    health : number
    minSal : number
    gasSal : number
    nrgSal : number
    abort : boolean
    commands : Array<string>

    constructor(loc: ALocation) {
        super()
        this.loc = loc
        this.occ = false
        this.use = false
        this.drive = false
        this.traverse = true
        this.owner = 0
        this.icon = "G"
        this.minCost = 0
        this.gasCost = 0
        this.nrgCost = 0
        this.health = 0
        this.minSal = 0
        this.gasSal = 0
        this.nrgSal = 0
        this.abort = false
        this.commands = new Array<string>("")
    }
}